static/app/views/insights/cache/components/samplePanel.tsx

```diff
@@ -229,6 +236,49 @@ export function CacheSamplePanel() {
     refetchCacheMisses();
   };
 
+  const avg =
+    transactionDurationData?.[0]?.[`avg(${MetricsFields.TRANSACTION_DURATION})`] ?? 0;
+
+  const samplesPlottable = useMemo(() => {
+    // Create a `TabularData` object from multiple datasets. This requires
+    // setting the meta manually. A little annoying, but the meta is simple.
+    const sampleData = {
+      data: spansWithDuration,
+      meta: {
+        fields: {'span.duration': 'duration'},
+        units: {'span.duration': DurationUnit.MILLISECOND},
+      },
```

Comment:
Can't you just combine the meta from the transaction response and the spans response? We that here
https://github.com/getsentry/sentry/blob/caee47f0ba23d234813b0f22757f2b370f69ade0/static/app/views/performance/cache/cacheLandingPage.tsx#L126 
You can probably just extract the `combineMeta` function and use it here too!

Reply:
Not yet, unfortunately! The meta types for `EventsMetaType` and for `TabularData` are misaligned right now. That's a good util though, I think there are a few other places I can use it.

Reply:
👍